File: app/src/main/java/com/mutualmobile/praxis/BaseApplication.java
package com.mutualmobile.praxis;

import android.app.Application;
import com.crashlytics.android.Crashlytics;
import com.mutualmobile.praxis.injection.component.ApplicationComponent;
import com.mutualmobile.praxis.injection.component.DaggerApplicationComponent;
import com.mutualmobile.praxis.injection.module.AppModule;
import io.fabric.sdk.android.Fabric;
import timber.log.Timber;

public class BaseApplication extends Application {

  private ApplicationComponent component;

  @Override public void onCreate() {
    super.onCreate();

    if (BuildConfig.DEBUG) {
      Timber.plant(new Timber.DebugTree());
    } else {
      Fabric.with(this, new Crashlytics());
    }
  }

  public ApplicationComponent getComponent() {
    if (component == null) {
      component = DaggerApplicationComponent.builder().appModule(new AppModule(this)).build();
    }
    return component;
  }
}

File: app/src/main/java/com/mutualmobile/praxis/BaseApplication.kt
package com.mutualmobile.praxis

import com.crashlytics.android.Crashlytics
import com.mutualmobile.praxis.injection.component.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import io.fabric.sdk.android.Fabric
import timber.log.Timber

class BaseApplication : DaggerApplication() {
  override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
    return DaggerAppComponent.builder().create(this)
  }

  override fun onCreate() {
    super.onCreate()

    if (BuildConfig.DEBUG) {
      Timber.plant(Timber.DebugTree())
    } else {
      Fabric.with(this, Crashlytics())
    }
  }

}
